File: qt/gen_qpoint.cpp
#include <QPoint>
#include <QPointF>
#include <qpoint.h>
#include "gen_qpoint.h"
#include "_cgo_export.h"

void QPoint_new(QPoint** outptr_QPoint) {
	QPoint* ret = new QPoint();
	*outptr_QPoint = ret;
}

void QPoint_new2(int xpos, int ypos, QPoint** outptr_QPoint) {
	QPoint* ret = new QPoint(static_cast<int>(xpos), static_cast<int>(ypos));
	*outptr_QPoint = ret;
}

void QPoint_new3(QPoint* param1, QPoint** outptr_QPoint) {
	QPoint* ret = new QPoint(*param1);
	*outptr_QPoint = ret;
}

bool QPoint_IsNull(const QPoint* self) {
	return self->isNull();
}

int QPoint_X(const QPoint* self) {
	return self->x();
}

int QPoint_Y(const QPoint* self) {
	return self->y();
}

void QPoint_SetX(QPoint* self, int x) {
	self->setX(static_cast<int>(x));
}

void QPoint_SetY(QPoint* self, int y) {
	self->setY(static_cast<int>(y));
}

int QPoint_ManhattanLength(const QPoint* self) {
	return self->manhattanLength();
}

QPoint* QPoint_Transposed(const QPoint* self) {
	return new QPoint(self->transposed());
}

QPoint* QPoint_OperatorPlusAssign(QPoint* self, QPoint* p) {
	QPoint& _ret = self->operator+=(*p);
	// Cast returned reference into pointer
	return &_ret;
}

QPoint* QPoint_OperatorMinusAssign(QPoint* self, QPoint* p) {
	QPoint& _ret = self->operator-=(*p);
	// Cast returned reference into pointer
	return &_ret;
}

QPoint* QPoint_OperatorMultiplyAssign(QPoint* self, float factor) {
	QPoint& _ret = self->operator*=(static_cast<float>(factor));
	// Cast returned reference into pointer
	return &_ret;
}

QPoint* QPoint_OperatorMultiplyAssignWithFactor(QPoint* self, double factor) {
	QPoint& _ret = self->operator*=(static_cast<double>(factor));
	// Cast returned reference into pointer
	return &_ret;
}

QPoint* QPoint_OperatorMultiplyAssign2(QPoint* self, int factor) {
	QPoint& _ret = self->operator*=(static_cast<int>(factor));
	// Cast returned reference into pointer
	return &_ret;
}

QPoint* QPoint_OperatorDivideAssign(QPoint* self, double divisor) {
	QPoint& _ret = self->operator/=(static_cast<qreal>(divisor));
	// Cast returned reference into pointer
	return &_ret;
}

int QPoint_DotProduct(QPoint* p1, QPoint* p2) {
	return QPoint::dotProduct(*p1, *p2);
}

void QPoint_Delete(QPoint* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QPoint*>( self );
	} else {
		delete self;
	}
}

void QPointF_new(QPointF** outptr_QPointF) {
	QPointF* ret = new QPointF();
	*outptr_QPointF = ret;
}

void QPointF_new2(QPoint* p, QPointF** outptr_QPointF) {
	QPointF* ret = new QPointF(*p);
	*outptr_QPointF = ret;
}

void QPointF_new3(double xpos, double ypos, QPointF** outptr_QPointF) {
	QPointF* ret = new QPointF(static_cast<qreal>(xpos), static_cast<qreal>(ypos));
	*outptr_QPointF = ret;
}

void QPointF_new4(QPointF* param1, QPointF** outptr_QPointF) {
	QPointF* ret = new QPointF(*param1);
	*outptr_QPointF = ret;
}

double QPointF_ManhattanLength(const QPointF* self) {
	qreal _ret = self->manhattanLength();
	return static_cast<double>(_ret);
}

bool QPointF_IsNull(const QPointF* self) {
	return self->isNull();
}

double QPointF_X(const QPointF* self) {
	qreal _ret = self->x();
	return static_cast<double>(_ret);
}

double QPointF_Y(const QPointF* self) {
	qreal _ret = self->y();
	return static_cast<double>(_ret);
}

void QPointF_SetX(QPointF* self, double x) {
	self->setX(static_cast<qreal>(x));
}

void QPointF_SetY(QPointF* self, double y) {
	self->setY(static_cast<qreal>(y));
}

QPointF* QPointF_Transposed(const QPointF* self) {
	return new QPointF(self->transposed());
}

QPointF* QPointF_OperatorPlusAssign(QPointF* self, QPointF* p) {
	QPointF& _ret = self->operator+=(*p);
	// Cast returned reference into pointer
	return &_ret;
}

QPointF* QPointF_OperatorMinusAssign(QPointF* self, QPointF* p) {
	QPointF& _ret = self->operator-=(*p);
	// Cast returned reference into pointer
	return &_ret;
}

QPointF* QPointF_OperatorMultiplyAssign(QPointF* self, double c) {
	QPointF& _ret = self->operator*=(static_cast<qreal>(c));
	// Cast returned reference into pointer
	return &_ret;
}

QPointF* QPointF_OperatorDivideAssign(QPointF* self, double c) {
	QPointF& _ret = self->operator/=(static_cast<qreal>(c));
	// Cast returned reference into pointer
	return &_ret;
}

double QPointF_DotProduct(QPointF* p1, QPointF* p2) {
	qreal _ret = QPointF::dotProduct(*p1, *p2);
	return static_cast<double>(_ret);
}

QPoint* QPointF_ToPoint(const QPointF* self) {
	return new QPoint(self->toPoint());
}

void QPointF_Delete(QPointF* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QPointF*>( self );
	} else {
		delete self;
	}
}

File: qt/gen_qpoint.go
package qt

import (
	"unsafe"
)

type QPoint struct {
	h          uintptr
	isSubclass bool
}

// NewQPoint constructs a new QPoint object.
func NewQPoint() *QPoint {
	g := newQPoint(QPoint_new())
	g.isSubclass = true
	return g
}

// NewQPoint2 constructs a new QPoint object.
func NewQPoint2(xpos int, ypos int) *QPoint {
	g := newQPoint(QPoint_new2((int)(xpos), (int)(ypos)))
	g.isSubclass = true
	return g
}

// NewQPoint3 constructs a new QPoint object.
func NewQPoint3(param1 *QPoint) *QPoint {
	g := newQPoint(QPoint_new3(param1.cPointer()))
	g.isSubclass = true
	return g
}

func (this *QPoint) IsNull() bool {
	return (bool)(QPoint_IsNull(this.h))
}

func (this *QPoint) X() int {
	return (int)(QPoint_X(this.h))
}

func (this *QPoint) Y() int {
	return (int)(QPoint_Y(this.h))
}

func (this *QPoint) SetX(x int) {
	QPoint_SetX(this.h, (int)(x))
}

func (this *QPoint) SetY(y int) {
	QPoint_SetY(this.h, (int)(y))
}

func (this *QPoint) ManhattanLength() int {
	return (int)(QPoint_ManhattanLength(this.h))
}

func (this *QPoint) Transposed() *QPoint {
	_goptr := newQPoint(QPoint_Transposed(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QPoint) OperatorPlusAssign(p *QPoint) *QPoint {
	return newQPoint(QPoint_OperatorPlusAssign(this.h, p.cPointer()))
}

func (this *QPoint) OperatorMinusAssign(p *QPoint) *QPoint {
	return newQPoint(QPoint_OperatorMinusAssign(this.h, p.cPointer()))
}

func (this *QPoint) OperatorMultiplyAssign(factor float32) *QPoint {
	return newQPoint(QPoint_OperatorMultiplyAssign(this.h, (float)(factor)))
}

func (this *QPoint) OperatorMultiplyAssignWithFactor(factor float64) *QPoint {
	return newQPoint(QPoint_OperatorMultiplyAssignWithFactor(this.h, (double)(factor)))
}

func (this *QPoint) OperatorMultiplyAssign2(factor int) *QPoint {
	return newQPoint(QPoint_OperatorMultiplyAssign2(this.h, (int)(factor)))
}

func (this *QPoint) OperatorDivideAssign(divisor float64) *QPoint {
	return newQPoint(QPoint_OperatorDivideAssign(this.h, (double)(divisor)))
}

func QPoint_DotProduct(p1 *QPoint, p2 *QPoint) int {
	return (int)(QPoint_DotProduct(p1.cPointer(), p2.cPointer()))
}

func (this *QPoint) ToPointF() *QPointF {
	_goptr := newQPointF(QPoint_ToPointF(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

type QPointF struct {
	h          uintptr
	isSubclass bool
}

// NewQPointF constructs a new QPointF object.
func NewQPointF() *QPointF {
	g := newQPointF(QPointF_new())
	g.isSubclass = true
	return g
}

// NewQPointF2 constructs a new QPointF object.
func NewQPointF2(p *QPoint) *QPointF {
	g := newQPointF(QPointF_new2(p.cPointer()))
	g.isSubclass = true
	return g
}

// NewQPointF3 constructs a new QPointF object.
func NewQPointF3(xpos float64, ypos float64) *QPointF {
	g := newQPointF(QPointF_new3((double)(xpos), (double)(ypos)))
	g.isSubclass = true
	return g
}

// NewQPointF4 constructs a new QPointF object.
func NewQPointF4(param1 *QPointF) *QPointF {
	g := newQPointF(QPointF_new4(param1.cPointer()))
	g.isSubclass = true
	return g
}

func (this *QPointF) ManhattanLength() float64 {
	return (float64)(QPointF_ManhattanLength(this.h))
}

func (this *QPointF) IsNull() bool {
	return (bool)(QPointF_IsNull(this.h))
}

func (this *QPointF) X() float64 {
	return (float64)(QPointF_X(this.h))
}

func (this *QPointF) Y() float64 {
	return (float64)(QPointF_Y(this.h))
}

func (this *QPointF) SetX(x float64) {
	QPointF_SetX(this.h, (double)(x))
}

func (this *QPointF) SetY(y float64) {
	QPointF_SetY(this.h, (double)(y))
}

func (this *QPointF) Transposed() *QPointF {
	_goptr := newQPointF(QPointF_Transposed(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QPointF) OperatorPlusAssign(p *QPointF) *QPointF {
	return newQPointF(QPointF_OperatorPlusAssign(this.h, p.cPointer()))
}

func (this *QPointF) OperatorMinusAssign(p *QPointF) *QPointF {
	return newQPointF(QPointF_OperatorMinusAssign(this.h, p.cPointer()))
}

func (this *QPointF) OperatorMultiplyAssign(c float64) *QPointF {
	return newQPointF(QPointF_OperatorMultiplyAssign(this.h, (double)(c)))
}

func (this *QPointF) OperatorDivideAssign(c float64) *QPointF {
	return newQPointF(QPointF_OperatorDivideAssign(this.h, (double)(c)))
}

func QPointF_DotProduct(p1 *QPointF, p2 *QPointF) float64 {
	return (float64)(QPointF_DotProduct(p1.cPointer(), p2.cPointer()))
}

func (this *QPointF) ToPoint() *QPoint {
	_goptr := newQPoint(QPointF_ToPoint(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}
